akka-stream-tests/src/test/scala/akka/stream/FusingSpec.scala

```diff
@@ -69,6 +69,47 @@ class FusingSpec extends StreamSpec {
       }
     }
 
+    "use one actor per grouped substream when there is an async boundary around the flow (manual)" in {
+      def ref = {
+        val bus = GraphInterpreter.currentInterpreter.log.asInstanceOf[BusLogging]
+        bus.logSource
+      }
+      val async = Flow[Int].map(x ⇒ { testActor ! ref; x }).async
+      Source(0 to 9)
+        .map(x ⇒ { testActor ! ref; x })
+        .groupBy(10, identity)
+        .via(async)
+        .mergeSubstreams
+        .runWith(Sink.seq)
+        .futureValue
+        .sorted should ===(0 to 9)
+      val refs = receiveN(10)
+      withClue(s"refs=\n${refs.mkString("\n")}") {
+        refs.toSet should have size (11) // main flow + 10 subflows
+      }
+    }
+
+    "use one actor per grouped substream when there is an async boundary around the flow (combinator)" in {
+      def ref = {
+        val bus = GraphInterpreter.currentInterpreter.log.asInstanceOf[BusLogging]
+        bus.logSource
+      }
+      val flow = Flow[Int].map(x ⇒ { testActor ! ref; x })
+      Source(0 to 9)
+        .map(x ⇒ { testActor ! ref; x })
+        .groupBy(10, identity)
+        .via(flow)
+        .async
+        .mergeSubstreams
+        .runWith(Sink.seq)
+        .futureValue
+        .sorted should ===(0 to 9)
+      val refs = receiveN(10)
+
+      refs.toSet should have size (11) // outer/main flow + 10 subflows
+
+    }
```

Comment:
this somehow is 10 actors, so one per substream but not a separate one for the stages outside the substream. Very strange.

Reply:
Actually today seeing a different behavior here, not sure if I misinterpreted the output before. Now I see one actor for the "main" flow and then each sub stream is added to the same async island.